src/sagemaker/mxnet/estimator.py

```diff
@@ -120,3 +247,63 @@ def _prepare_init_params_from_job_description(cls, job_details):
+        Args:
+            inputs (str or dict or sagemaker.session.s3_input): Information about the training data.
+                This can be one of three types:
+                (str) - the S3 location where training data is saved.
+                (dict[str, str] or dict[str, sagemaker.session.s3_input]) - If using multiple channels for
+                    training data, you can specify a dict mapping channel names
+                    to strings or :func:`~sagemaker.session.s3_input` objects.
+                (sagemaker.session.s3_input) - channel configuration for S3 data sources that can provide
+                    additional information as well as the path to the training dataset.
+                    See :func:`sagemaker.session.s3_input` for full details.
+            wait (bool): Whether the call should wait until the job completes (default: True).
+            logs (bool): Whether to show the logs produced by the job.
+                Only meaningful when wait is True (default: True).
+            job_name (str): Training job name. If not specified, the estimator generates a default job name,
+                based on the training image name and current timestamp.
+            run_tensorboard_locally (bool): Whether to execute TensorBoard in a different process with
+                downloaded checkpoint information (default: False). This is an experimental feature, and requires
+                TensorBoard and AWS CLI to be installed. It terminates TensorBoard when execution ends.
+        """
+        def fit_super():
+            super(MXNet, self).fit(inputs, wait, logs, job_name)
+
+        if run_tensorboard_locally and wait is False:
+            raise ValueError("Tensorboard is not supported with async fit")
+
+        if run_tensorboard_locally:
+            tensorboard = Tensorboard(self)
+            tensorboard.validate_requirements()
+            try:
+                tensorboard.start()
+                fit_super()
+            finally:
+                # sleep 20 secs for tensorboard start up if fit() quits instantly
+                time.sleep(20)
+                tensorboard.event.set()
+                tensorboard.join()
+        else:
+            fit_super()        
+
+    def hyperparameters(self):
+        """Return hyperparameters used by your custom TensorFlow code during model training."""
+        hyperparameters = super(MXNet, self).hyperparameters()
+
+        if not self.checkpoint_path:
+            local_code = get_config_value('local.local_code', self.sagemaker_session.config)
+            if self.sagemaker_session.local_mode and local_code:
+                self.checkpoint_path = '/opt/ml/shared/checkpoints'
+            else:
+                self.checkpoint_path = os.path.join(self.output_path,
+                                                    self._current_job_name, 'checkpoints')
+
+
+        additional_hyperparameters = {'checkpoint_path': self.checkpoint_path}
```

Comment:
I'm a little confused how this works -- the TensorFlow container uses checkpoint_path to write checkpoints to S3, which checkpoints TensorBoard reads, but the MXNet container doesn't do anything with checkpoint_path. How does the MXNet container write checkpoints, and how does the Python SDK know how to point TensorBoard to those checkpoints?

Reply:
OK, thanks.

So it's up to the customer to write these checkpoints to the right place in S3, which is a pretty different experience compared to TF (which handles writing checkpoints to the right place in S3 during training).

I think this would be a good feature in the MXNet container, but it seems premature to me to put this in the Python SDK before it's supported in the MXNet container.